src/lib/dai.c

```diff
@@ -162,7 +198,8 @@ const struct device *dai_get_device(uint32_t type, uint32_t index)
 	for (i = 0; i < ARRAY_SIZE(zephyr_dev); i++) {
 		if (dai_config_get(zephyr_dev[i], &cfg, dir))
 			continue;
-		if (cfg.type == type && cfg.dai_index == index)
+		z_type = dai_type_to_zephyr(type);
```

Comment:
no need to do this in the `for` loop. Just do it before.

also, maybe an error message if the conversion fails?